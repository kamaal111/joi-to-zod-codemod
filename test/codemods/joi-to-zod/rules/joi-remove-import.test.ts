import { test, expect } from 'vitest';
import { Lang, parseAsync } from '@ast-grep/napi';

import joiRemoveImport from '../../../../src/codemods/joi-to-zod/rules/joi-remove-import';

test('Joi remove import', async () => {
  const source = `
import Joi from 'joi';

enum Job {
  Developer = 'developer',
  DevOps = 'devops',
  Designer = 'designer',
}

export const employee = Joi.object().keys({
  job: Joi.string().valid(...Object.values(Job)),
});
`;
  const ast = await parseAsync(Lang.TypeScript, source);

  const modifications = await joiRemoveImport({
    ast,
    report: { changesApplied: 0 },
    lang: Lang.TypeScript,
    filename: null,
    history: [ast],
  });
  const updatedSource = modifications.ast.root().text();

  expect(modifications.report.changesApplied).toBe(1);
  expect(modifications.history.length).toBe(2);
  expect(updatedSource).not.toContain("import Joi from 'joi';");
  expect(updatedSource).toMatchSnapshot();
});
