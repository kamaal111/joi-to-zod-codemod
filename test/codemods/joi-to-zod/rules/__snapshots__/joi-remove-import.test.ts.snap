// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Joi remove import 1`] = `
"enum Job {
  Developer = 'developer',
  DevOps = 'devops',
  Designer = 'designer',
}

export const employee = Joi.object().keys({
  job: Joi.string().valid(...Object.values(Job)),
});
"
`;
